### PR TITLE
Suggestions bug fix

### LINKED ISSUE #3: Fix suggestions for Select input
- `const selectSuggestions = _.uniqWith(this.state.suggestions, _.isEqual)` 
When channel and campaign are the same with this solution it will remove one of them.

- ```props.dataCSV.data.forEach((element, i) => {
      this.setState((state, props) => ({
        suggestions: [
          ...state.suggestions,
          { label: props.dataCSV.data[i].campaign },
          { label: props.dataCSV.data[i].channel }
        ]
      }));
 It could be better to have suggestions for channels and campaigns separately.

### PR DESCRIPTION
**Pull request by zeljko-djuric**
Fixes #3

### CODE REVIEW

File: src/components/DisplayMetrics.js [hunk 1]
```diff
@@ -68,7 +72,15 @@ class DisplayMetrics extends React.Component {
...
     const numberOfImpresions = this.state.impressions;
-    const selectSuggestions = _.uniqWith(this.state.suggestions, _.isEqual);
+    const uniqueCampaign = _.uniqWith(
+      this.state.suggestionsForCampaign,
+      _.isEqual
+    );
+    const uniqeChannel = _.uniqWith(
+      this.state.suggestionsForChannel,
+      _.isEqual
+    );
```
Comment (vlabs) [inline src/components/DisplayMetrics.js]: don't repeat youself DRY 

please extract this to a function (eg `uniqueObjects`) and place this in `src/utils`

File: src/components/DisplayMetrics.js [hunk 2]
```diff
@@ -68,7 +72,15 @@ class DisplayMetrics extends React.Component {
...
+    const selectSuggestions = _.concat(uniqeChannel, uniqueCampaign);
```
Comment (vlabs) [inline src/components/DisplayMetrics.js]: i would also place this into a utils function, to separate business logic from the view

File: src/components/DisplayMetrics.js [hunk 3]
```diff
@@ -15,9 +16,12 @@ class DisplayMetrics extends React.Component {
...
-        suggestions: [
-          ...state.suggestions,
-          { label: props.dataCSV.data[i].campaign },
+        suggestionsForCampaign: [
+          ...state.suggestionsForCampaign,
+          { label: props.dataCSV.data[i].campaign }
+        ],
+        suggestionsForChannel: [
+          ...state.suggestionsForChannel,
           { label: props.dataCSV.data[i].channel }
```
Comment (vlabs) [inline src/components/DisplayMetrics.js:27]: DRY 

i would also place this into a utils function, to separate business logic from the view

**Review by vlabs — CHANGES_REQUESTED** [review]
you can explicitly request a review from me but note that i am 3 weeks on vacation now

File: src/utils/uniqueObjects.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import _ from "lodash";
+
+const uniqueObjects = (campaign, channel) => {
```
Comment (vlabs) [inline src/utils/uniqueObjects.js]: can be simplified to 
```
const uniqueObjects = (campaign, channel) => _.concat(_.uniqWith(campaign, _.isEqual), _.uniqWith(channel, _.isEqual))
```

**Review by vlabs — CHANGES_REQUESTED** [review]

**Comment by vlabs** [comment]
and please answer change request with the commit hash, so we can backtrack changes

File: src/utils/uniqueObjects.js (hunk 4 above)
Comment (vlabs) [reply src/utils/uniqueObjects.js]: nice, and you can also get rid of the `return` statement, see above

Comment (zeljko-djuric) [reply src/utils/uniqueObjects.js]: 9c48c4a2e684e8069e1118a0950ed72a162cdf26

**Comment by vlabs** [comment]
you have conflicts. merge develop into this branch and resolve them

File: src/utils/uniqueObjects.js (hunk 4 above)
Comment (vlabs) [reply src/utils/uniqueObjects.js]: :+1:

**Comment by vlabs** [comment]
there are still three requests without an answer. please implement/answer with commit hash

File: src/components/DisplayMetrics.js (hunk 1 above)
Comment (zeljko-djuric) [reply src/components/DisplayMetrics.js]: 648eb83d51cb96ab87ac8c5d2f4df8a04c2c1336

File: src/components/DisplayMetrics.js (hunk 2 above)
Comment (zeljko-djuric) [reply src/components/DisplayMetrics.js]: 648eb83d51cb96ab87ac8c5d2f4df8a04c2c1336 with this change, I also placed previous concatanation into a utils function.

**Review by vlabs — APPROVED** [review]
let's merge this

**Pull request merged**